persist/fs/index_lookup_prop_test.go

```diff
@@ -0,0 +1,240 @@
+	// Define property test function which will be passed various propTestInputs
+	propertyFunc := func(input propTestInput) (bool, error) {
+		// Filter out duplicate IDs
+		writes := []generatedWrite{}
+		unique := map[string]struct{}{}
+		for _, write := range input.realWrites {
+			s := string(write.id.Data().Get())
+			if _, ok := unique[s]; ok {
+				continue
+			}
+			unique[s] = struct{}{}
+			writes = append(writes, write)
+		}
+
+		// Create a temporary directory for each test run
+		dir, err := ioutil.TempDir("", "testdb")
+		if err != nil {
+			return false, err
+		}
+		filePathPrefix := filepath.Join(dir, "")
+		defer os.RemoveAll(dir)
+
+		options := NewOptions().
+			// Make sure that every index entry is also in the summaries file for the
+			// sake of verifying behavior
+			SetIndexSummariesPercent(1).
+			SetFilePathPrefix(filePathPrefix).
+			SetWriterBufferSize(testWriterBufferSize)
+		shard := input.shard
+
+		// Instantiate a writer and write the test data
+		w, err := NewWriter(options)
+		if err != nil {
+			return false, fmt.Errorf("err creating writer: %v, ", err)
+		}
+		err = w.Open(testNs1ID, testBlockSize, shard, testWriterStart)
+		if err != nil {
+			return false, fmt.Errorf("err opening writer: %v, ", err)
+		}
+		shardDirPath := ShardDirPath(filePathPrefix, testNs1ID, shard)
+		err = writeTestSummariesData(w, writes)
+		if err != nil {
+			return false, fmt.Errorf("err writing test summaries data: %v, ", err)
+		}
+
+		// Figure out the offsets for the writes so we have something to compare
+		// our results against
+		expectedIndexFileOffsets, err := readIndexFileOffsets(
+			shardDirPath, len(writes), testWriterStart)
+		if err != nil {
+			return false, fmt.Errorf("err reading index file offsets: %v", err)
+		}
+
+		// Read the summaries file into memory
+		summariesFilePath := filesetPathFromTime(
+			shardDirPath, testWriterStart, summariesFileSuffix)
+		summariesFile, err := os.Open(summariesFilePath)
+		if err != nil {
+			return false, fmt.Errorf("err opening summaries file: %v, ", err)
+		}
+		summariesFdWithDigest := digest.NewFdWithDigestReader(options.InfoReaderBufferSize())
+		expectedSummariesDigest := calculateExpectedChecksum(t, summariesFilePath)
+		decoder := msgpack.NewDecoder(options.DecodingOptions())
+		indexLookup, err := readNearestIndexOffsetLookupFromSummaries(
+			summariesFile, summariesFdWithDigest, expectedSummariesDigest, decoder, len(writes))
+		if err != nil {
+			return false, fmt.Errorf("err reading index lookup from summaries file: %v, ", err)
+		}
+
+		// Make sure it returns the correct index offset for every ID
+		for id, expectedOffset := range expectedIndexFileOffsets {
+			foundOffset, ok, err := indexLookup.getNearestIndexFileOffset(ts.StringID(id))
+			if err != nil {
+				return false, fmt.Errorf("Err locating index file offset for: %s, err: %v", id, err)
+			}
+			if !ok {
+				return false, fmt.Errorf("Unable to locate index file offset for: %s", id)
```

Comment:
nit: Lowercase the first char of the error (similar to our log output naming convention)

Reply:
done